libvast/vast/detail/actor_cast_wrapper.hpp

```diff
@@ -0,0 +1,31 @@
+/******************************************************************************
+ *                    _   _____   __________                                  *
+ *                   | | / / _ | / __/_  __/     Visibility                   *
+ *                   | |/ / __ |_\ \  / /          Across                     *
+ *                   |___/_/ |_/___/ /_/       Space and Time                 *
+ *                                                                            *
+ * This file is part of VAST. It is subject to the license terms in the       *
+ * LICENSE file found in the top-level directory of this distribution and at  *
+ * http://vast.io/license. No part of VAST, including this file, may be       *
+ * copied, modified, propagated, or distributed except according to the terms *
+ * contained in the LICENSE file.                                             *
+ ******************************************************************************/
+
+#pragma once
+
+#include "vast/fwd.hpp"
+
+#include <caf/actor_cast.hpp>
+
+#include <utility>
+
+namespace vast::detail {
+
+struct actor_cast_wrapper {
```

Comment:
Can you briefly explain in a comment what this does and/or give an example?

Reply:
```cpp
actor_cast_wrapper{}
```

is essentially this from C++20:

```cpp
[]<typename Out>(auto&& in) {
  return caf::actor_cast<Out>(
    std::forward<decltype(in)>(in));
}
```

I'll add that as a comment with a note to replace once switching to C++20.

Reply:
It's quite annoying that we have to add those one-offs because of a lack of orthogonality in the language.

Reply:
Even more so because these cannot be defined anonymously within functions. 😤